xds/src/main/java/io/grpc/xds/XdsLrsClient.java

```diff
@@ -0,0 +1,325 @@
+      stopwatch.reset().start();
+      // TODO: get load report and send to management server
+      lrsRequestWriter.onNext(LoadStatsRequest.newBuilder()
+          .setNode(Node.newBuilder()
+              .setMetadata(Struct.newBuilder()
+                  .putFields(
+                      TRAFFICDIRECTOR_GRPC_HOSTNAME,
+                      Value.newBuilder().setStringValue(serviceName).build())))
+          .addClusterStats(ClusterStats.newBuilder().setLoadReportInterval(
+              numToDuration(interval, TimeUnit.NANOSECONDS)))
+          .build());
+      scheduleNextLoadReport();
+    }
+
+    private void scheduleNextLoadReport() {
+      if (loadReportIntervalNano > 0) {
+        // Cancel pending load report and reschedule with updated load reporting interval.
+        if (loadReportTimer != null && loadReportTimer.isPending()) {
+          loadReportTimer.cancel();
+        }
+        loadReportTimer = syncContext.schedule(
+            new LoadReportingTask(this), loadReportIntervalNano, TimeUnit.NANOSECONDS,
+            timerService);
+      }
+    }
+
+    private void handleResponse(LoadStatsResponse response) {
+      if (closed) {
+        return;
+      }
+      logger.log(ChannelLogLevel.DEBUG, "Got an LRS response: {0}", response);
+
+      if (!initialResponseReceived) {
+        initialResponseReceived = true;
+      }
+      loadReportIntervalNano
+          = durationToNum(response.getLoadReportingInterval(), TimeUnit.NANOSECONDS);
+      List<String> serviceList = Collections.unmodifiableList(response.getClustersList());
+      // For gRPC use case, LRS response will only contain one cluster, which is the same as in
+      // the EDS response.
+      checkState(serviceList.size() == 1, "Excessive reporting cluster: %s", serviceList);
+      checkState(serviceList.get(0).equals(serviceName), "Unmatched cluster name: %s with EDS: %s",
+          serviceList.get(0), serviceName);
+      scheduleNextLoadReport();
+    }
+
+    private void handleStreamClosed(Status status) {
+      checkArgument(!status.isOk(), "unexpected OK status");
+      if (closed) {
+        return;
+      }
+      closed = true;
+      cleanUp();
+
+      long delayNanos = 0;
+      if (initialResponseReceived || lrsRpcRetryPolicy == null) {
+        // Reset the backoff sequence if balancer has sent the initial response, or backoff sequence
+        // has never been initialized.
+        lrsRpcRetryPolicy = backoffPolicyProvider.get();
+      }
+      // Backoff only when balancer wasn't working previously.
+      if (!initialResponseReceived) {
+        // The back-off policy determines the interval between consecutive RPC upstarts, thus the
+        // actual delay may be smaller than the value from the back-off policy, or even negative,
+        // depending how much time was spent in the previous RPC.
+        delayNanos =
+            lrsRpcRetryPolicy.nextBackoffNanos() - stopwatch.elapsed(TimeUnit.NANOSECONDS);
```

Comment:
Using `stopwatch` here is technically correct, because if `initialResponseReceived == false`, `sendLoadReport()` must not have been called thus `stopwatch` still points at the time where the stream is started.

However, I would prefer using a separate stopwatch for different purposes, so that it's easier to reason about and less likely to cause bugs with code changes that may potentially break the premise. So instead of `stopwatch`, I would use `reportStopwatch` and `retryStopwatch`.

Reply:
Oh! This is a typo bug. It should be `lrsRpcRetryPolicy.nextBackoffNanos() - XdsLrsClient.this.stopwatch.elapsed(TimeUnit.NANOSECONDS);` which is `XdsLrsClient`'s stopwatch. Yes, having two stopwatches are confusing, I will change the names.